src/client/Microsoft.Identity.Client/Internal/Requests/IntegratedWindowsAuthRequest.cs

```diff
@@ -38,9 +38,22 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
             await ResolveAuthorityAsync().ConfigureAwait(false);
             await UpdateUsernameAsync().ConfigureAwait(false);
             var userAssertion = await FetchAssertionFromWsTrustAsync().ConfigureAwait(false);
-            var msalTokenResponse = await SendTokenRequestAsync(
-                                            GetAdditionalBodyParameters(userAssertion), cancellationToken)
-                                        .ConfigureAwait(false);
+
+            MsalTokenResponse msalTokenResponse;
+            try
+            {
+                msalTokenResponse = await SendTokenRequestAsync(
+                                                GetAdditionalBodyParameters(userAssertion), cancellationToken)
+                                            .ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                throw new MsalClientException(
```

Comment:
Given the error happens when sending the token request, shouldn't it be an MsalServiceException?

Reply:
Moving this to a new location where only client requests will be thrown. This try catch will be removed